other/secrets-not-from-env-vars-cve-2024-3177/secrets-not-from-env-vars-cve-2024-3177.yaml

```diff
@@ -0,0 +1,44 @@
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: secrets-not-from-env-vars-cve-2024-3177
+  annotations:
+    policies.kyverno.io/title: Prevent CVE-2024-3177
+    policies.kyverno.io/category: Security
+    policies.kyverno.io/severity: low
+    policies.kyverno.io/subject: Pod, Secret, envFrom
+    kyverno.io/kubernetes-version: 1.27
+    kyverno.io/kyverno-version: 1.11.0
+    policies.kyverno.io/description: >-
+      Secrets used as environment variables containing sensitive information may, if not carefully controlled,
+      be printed in log output which could be visible to unauthorized people and captured in forwarding
+      applications. This policy specifically blocks CVE-2024-3177
+spec:
+  validationFailureAction: Audit
+  background: true
+  rules:
+  - name: secrets-not-from-envfrom-cve-2024-3177
+    match:
+      any:
+      - resources:
+          kinds:
+          - Pod
+    validate:
+      message: "Secrets must not come from envFrom statements. CVE-2024-3177"
+      pattern:
+        metadata:
+          annotations:
+            kubernetes.io/enforce-mountable-secrets: "true"
```

Comment:
Should this be a conditional check? I.e., if this annotation is set to true, then the pattern must be satisfied?

Reply:
sure will fix

Reply:
Do we need the annotation at all? 

Seems to me that the policy would be more widely applicable without it.